### PR TITLE
More CLI integration tests

### PR DESCRIPTION
**Pull request by iketheadore**
* Test `createRawTransaction` command
* Test `broadcastRawTransaction` command
* Test `walletHistory` command
* Test `checkdb` command
* Test `version` command
* Test `generateWallet` command

refs: #939

### CODE REVIEW

File: src/api/cli/integration/integration_test.go [hunk 1]
```diff
@@ -1523,8 +1519,186 @@ func TestLiveSend(t *testing.T) {
 	}
 }
 
+// TestLiveCreateAndBroadcastRawTransaction does almost the same procedure as TestLiveSend.
+// Create raw transaction with command arguments the same as TestLiveSend, then broadcast the
+// created raw transaction. After the transaction is confirmed, run the same transaction check
+// function like it in TestLiveSend.
```
Comment (therealssj) [inline src/api/cli/integration/integration_test.go]: like in

File: src/api/cli/integration/integration_test.go [hunk 2]
```diff
@@ -1523,8 +1519,186 @@ func TestLiveSend(t *testing.T) {
...
+		})
+	}
+}
+
 func getTransaction(t *testing.T, txid string) *webrpc.TxnResult {
 	output, err := exec.Command(binaryPath, "transaction", txid).CombinedOutput()
+	if err != nil {
+		fmt.Println(string(output))
+		return &webrpc.TxnResult{}
+	}
```
Comment (therealssj) [inline src/api/cli/integration/integration_test.go]: fail the test only?

File: src/gui/integration/integration_test.go [hunk 3]
```diff
@@ -1796,3 +1797,150 @@ func TestLivePendingTransactions(t *testing.T) {
...
+				// Confirms the address balance are equal to the totoalCoins
+				coins, _ = getAddressBalance(t, w.Entries[0].Address.String())
+				require.Equal(t, totalCoins, coins)
+			},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			c := gui.NewClient(nodeAddress())
```
Comment (therealssj) [inline src/gui/integration/integration_test.go]: declare out of `for` scope and use in subsequent functions

File: src/gui/integration/integration_test.go [hunk 4]
```diff
@@ -1796,3 +1797,150 @@ func TestLivePendingTransactions(t *testing.T) {
...
+// getAddressBalance gets balance of given address.
+// Returns coins and coin hours.
+func getAddressBalance(t *testing.T, addr string) (uint64, uint64) {
+	c := gui.NewClient(nodeAddress())
+	bp, err := c.Balance([]string{addr})
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
+	return bp.Confirmed.Coins, bp.Confirmed.Hours
+}
```
Comment (therealssj) [inline src/gui/integration/integration_test.go]: refactor functions using `gui.NewClient` to take that as an input parameter

File: src/gui/integration/integration_test.go [hunk 5]
```diff
@@ -1796,3 +1797,150 @@ func TestLivePendingTransactions(t *testing.T) {
...
+		t.Fatal("Missing WALLET_NAME environment value")
+	}
+
+	return walletDir, walletName
+}
+
+// getWalletBalance gets wallet balance.
+// Returns coins and hours
+func getWalletBalance(t *testing.T, walletName string) (uint64, uint64) {
+	c := gui.NewClient(nodeAddress())
```
Comment (therealssj) [inline src/gui/integration/integration_test.go]: refactor functions using `gui.NewClient` to take that as an input parameter

File: src/api/cli/integration/integration_test.go (hunk 2 above)
Comment (iketheadore) [reply src/api/cli/integration/integration_test.go]: No, we don't fail here, It's the error that the transaction doesn't exist, we only need to check again next time.

Comment (therealssj) [reply src/api/cli/integration/integration_test.go]: Oh okay 👌

File: src/api/cli/integration/integration_test.go [hunk 6]
```diff
@@ -1648,3 +1822,111 @@ func getWalletOutputs(t *testing.T, walletPath string) visor.ReadableOutputs {
...
+	output, err := exec.Command(binaryPath, "walletHistory").CombinedOutput()
+	require.NoError(t, err)
+	var his []cli.AddrHistory
+	err = json.NewDecoder(bytes.NewReader(output)).Decode(&his)
+	require.NoError(t, err)
+	// Confirms that the wallet has history.
+	require.True(t, len(his) > 0)
+}
+
+func TestCheckDB(t *testing.T) {
```
Comment (gz-c) [inline src/api/cli/integration/integration_test.go]: Is it possible for `TestLiveCheckDB` to check the user's db?

Comment (iketheadore) [reply src/api/cli/integration/integration_test.go]: Yes, I will add it.

Comment (iketheadore) [reply src/api/cli/integration/integration_test.go]: Hmm, seems can't. The live test loads database with writing access, we can't open it in another progress even try with readonly option.

Comment (gz-c) [reply src/api/cli/integration/integration_test.go]: oh, it would have to be a separate test with no node running, its fine

Comment (gz-c) [reply src/api/cli/integration/integration_test.go]: We should make a pre-release testing checklist, with instructions to run the live tests, and an additional step to run checkdb on a synced database

**Pull request merged**